tests/acceptance/test_mender-connect.py

```diff
@@ -0,0 +1,349 @@
+            proc_server_ws_two = None
+
+            # start the mocks
+            put_mock_files(connection)
+            proc_server_dbus = start_mock_dbus_server(connection)
+            proc_server_ws_one = start_mock_websockets_server(connection, 5000)
+            proc_server_ws_two = start_mock_websockets_server(connection, 6000)
+
+            wait_for_mock_dbus_server(connection)
+            dbus_set_token_and_url(
+                connection, "whateversecuretoken", "http://localhost:5000"
+            )
+
+            # start the mender-connect service
+            startup_time = time.time()
+            connection.run(
+                "systemctl --job-mode=ignore-dependencies start mender-connect"
+            )
+
+            # wait for first connect
+            _ = wait_for_string_in_log(
+                connection,
+                startup_time,
+                30,
+                "Connection established with http://localhost:5000",
+            )
+
+            # 1. Change token
+            signal_time = time.time()
+            dbus_set_token_and_url_and_emit_signal(
+                connection, "newtoken", "http://localhost:5000"
+            )
+            _ = wait_for_string_in_log(
+                connection,
+                signal_time,
+                30,
+                "Connection established with http://localhost:5000",
+            )
+
+            # 2. Change url
+            signal_time = time.time()
+            dbus_set_token_and_url_and_emit_signal(
+                connection, "newtoken", "http://localhost:6000"
+            )
+            _ = wait_for_string_in_log(
+                connection,
+                signal_time,
+                30,
+                "Connection established with http://localhost:6000",
+            )
+
+            # 3. Change token and url
+            signal_time = time.time()
+            dbus_set_token_and_url_and_emit_signal(
+                connection, "newtokenagainverysecure", "http://localhost:5000"
+            )
+            _ = wait_for_string_in_log(
+                connection,
+                signal_time,
+                30,
+                "Connection established with http://localhost:5000",
+            )
+
+            # 4. Unauthorize and re-authorize
+            signal_time = time.time()
+            dbus_set_token_and_url_and_emit_signal(connection, "", "")
+            _ = wait_for_string_in_log(
+                connection,
+                signal_time,
+                30,
+                "dbusEventLoop terminated 0 sessions, 0 shells",
+            )
+            dbus_set_token_and_url_and_emit_signal(
+                connection, "reauthorizedtoken", "http://localhost:6000"
+            )
+            _ = wait_for_string_in_log(
+                connection,
+                signal_time,
+                30,
+                "Connection established with http://localhost:6000",
+            )
+
+        finally:
+            connection.run(
```

Comment:
Suggestion: Seeing this pattern multiple places, how about a fixture?

Reply:
Good idea! I'll move into a fixture:
* installing the mock files
* starting the servers
* cleaning when done

:+1: